Doc/library/multiprocessing.rst

```diff
@@ -837,6 +837,10 @@ For an example of the usage of queues for interprocess communication see
       Return ``True`` if the queue is empty, ``False`` otherwise.  Because of
       multithreading/multiprocessing semantics, this is not reliable.
 
+      If :meth:`put` or :meth:`put_nowait` has been called previously (i.e.,
```

Comment:
Rather than attempt to specify _exactly_ what internal state can cause an OSError when, from a user stand point, just simplifying the added text to just state "This method may raise an :exc:`OSError` if the queue has been closed" should be enough for the docs.

The key thing to communicate is that it is not guaranteed to do so in _all_ circumstances (ie: this API is also not a reliable way to determine if the queue has been closed).  Thus a word like "may" "can" or "might" rather than an implied or stated "will".

Nobody using the API should try to replicate knowing the state, they just need to be prepared for the exception regardless and understand what it could mean.

Reply:
You are definitely right! I wanted to highlight the subtle difference with `SimpleQueue.empty` but it is better to have a documentation which does not implicitly encouraged incorrect usage.